### PR TITLE
Make Control Property Editor aware which application (scenario) its running in the iframe

### PR DESCRIPTION
**Pull request by nikmace**
Feat for #1414.

- Adds scenario to Control Property Editor.

### CODE REVIEW

File: .changeset/dull-spoons-clap.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@sap-ux-private/control-property-editor-common': minor
+'@sap-ux-private/preview-middleware-client': minor
+'@sap-ux/control-property-editor': minor
+---
```
Comment (zdravko-georgiev) [inline .changeset/dull-spoons-clap.md:5]: Changeset for `preview-middleware` is missing. Please add. Also for this change I believe that a `patch` update is sufficient.

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx [hunk 2]
```diff
@@ -104,6 +104,7 @@ describe('OutlinePanel', () => {
             scale: 1,
             outline: model,
             filterQuery: filterInitOptions,
+            scenario: 'UI_ADAPTATION',
```
Comment (zdravko-georgiev) [inline packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx:107]: Could you please use the `scenario` const from `@sap-ux-private/control-property-editor-common`?

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx [hunk 3]
```diff
@@ -138,6 +139,7 @@ describe('OutlinePanel', () => {
...
```
Comment (zdravko-georgiev) [inline packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx:142]: same as above

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx [hunk 4]
```diff
@@ -232,21 +235,14 @@ describe('OutlinePanel', () => {
...
-                editable: true,
-                visible: true
             }
         ];
         const initialState: State = {
             deviceType: DeviceType.Desktop,
             scale: 1,
             outline: model,
             filterQuery: filterInitOptions,
+            scenario: 'UI_ADAPTATION',
```
Comment (zdravko-georgiev) [inline packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx:245]: same as above

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx [hunk 5]
```diff
@@ -288,6 +284,7 @@ describe('OutlinePanel', () => {
             scale: 1,
             outline: model,
             filterQuery: filterInitOptions,
+            scenario: 'ADAPTATION_PROJECT',
```
Comment (zdravko-georgiev) [inline packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx:287]: same as above

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx [hunk 6]
```diff
@@ -331,6 +328,7 @@ describe('OutlinePanel', () => {
...
```
Comment (zdravko-georgiev) [inline packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx:331]: same as above

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx [hunk 7]
```diff
@@ -418,6 +416,7 @@ describe('OutlinePanel', () => {
             scale: 1,
             outline: getModel(true, true, true, true),
             filterQuery: filterInitOptions,
+            scenario: 'UI_ADAPTATION',
```
Comment (zdravko-georgiev) [inline packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx:419]: same as above

File: packages/control-property-editor/src/panels/outline/Tree.tsx [hunk 8]
```diff
@@ -358,43 +359,66 @@ export const Tree = (): ReactElement => {
...
-                    className={`tree-cell ${isExtensionPoint ? 'tooltip-container' : ''}`}
-                    onContextMenu={(e) => isExtensionPoint && handleOpenTooltip(e, tooltipId)}>
-                    {groupHeaderProps?.group?.count !== 0 && (
-                        <Icon
-                            className={`${chevronTransform}`}
-                            iconName={UiIcons.Chevron}
-                            onClick={(event) => {
-                                onToggleCollapse(groupHeaderProps);
-                                event.stopPropagation();
+                {scenario !== 'ADAPTATION_PROJECT' ? (
```
Comment (zdravko-georgiev) [inline packages/control-property-editor/src/panels/outline/Tree.tsx:362]: Could you please use the `scenario` const from `@sap-ux-private/control-property-editor-common`?

File: packages/preview-middleware-client/test/unit/cpe/init.test.ts [hunk 9]
```diff
@@ -69,7 +69,7 @@ describe('main', () => {
         attachSelectionChange,
         getSelection: jest.fn().mockReturnValue([{ setSelected: jest.fn() }, { setSelected: jest.fn() }]),
         attachUndoRedoStackModified: jest.fn(),
-        getFlexSettings: jest.fn().mockReturnValue({ layer: 'VENDOR' }),
+        getFlexSettings: jest.fn().mockReturnValue({ layer: 'VENDOR', scenario: 'UiAdaptation' }),
```
Comment (zdravko-georgiev) [inline packages/preview-middleware-client/test/unit/cpe/init.test.ts:72]: Could you please use the `scenario` const from `@sap-ux-private/control-property-editor-common`?

**Review by zdravko-georgiev — CHANGES_REQUESTED** [review]

File: packages/control-property-editor/src/panels/outline/Tree.tsx (hunk 8 above)
Comment (nikmace) [reply packages/control-property-editor/src/panels/outline/Tree.tsx:362]: This one here should not be a constant since it is coming from redux store.

Comment (zdravko-georgiev) [reply packages/control-property-editor/src/panels/outline/Tree.tsx:362]: Right 👍

File: packages/preview-middleware-client/test/unit/cpe/init.test.ts (hunk 9 above)
Comment (nikmace) [reply packages/preview-middleware-client/test/unit/cpe/init.test.ts]: Changed in: https://github.com/SAP/open-ux-tools/pull/1422/commits/83134d16a0fff1526e4dddf61c067912f7a5bdf2

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx (hunk 7 above)
Comment (nikmace) [reply packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx]: Changed in: https://github.com/SAP/open-ux-tools/pull/1422/commits/83134d16a0fff1526e4dddf61c067912f7a5bdf2

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx (hunk 6 above)
Comment (nikmace) [reply packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx]: Changed in: https://github.com/SAP/open-ux-tools/pull/1422/commits/83134d16a0fff1526e4dddf61c067912f7a5bdf2

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx (hunk 5 above)
Comment (nikmace) [reply packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx]: Changed in: https://github.com/SAP/open-ux-tools/pull/1422/commits/83134d16a0fff1526e4dddf61c067912f7a5bdf2

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx (hunk 4 above)
Comment (nikmace) [reply packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx]: Changed in: https://github.com/SAP/open-ux-tools/pull/1422/commits/83134d16a0fff1526e4dddf61c067912f7a5bdf2

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx (hunk 3 above)
Comment (nikmace) [reply packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx:142]: Changed in: https://github.com/SAP/open-ux-tools/pull/1422/commits/83134d16a0fff1526e4dddf61c067912f7a5bdf2

File: packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx (hunk 2 above)
Comment (nikmace) [reply packages/control-property-editor/test/unit/panels/outline/OutlinePanel.test.tsx]: Changed in: https://github.com/SAP/open-ux-tools/pull/1422/commits/83134d16a0fff1526e4dddf61c067912f7a5bdf2

File: .changeset/dull-spoons-clap.md (hunk 1 above)
Comment (nikmace) [reply .changeset/dull-spoons-clap.md]: Added in: https://github.com/SAP/open-ux-tools/pull/1422/commits/9fccc9128640cb8f8dacebb1089b45fbdaef7768

File: .changeset/dull-spoons-clap.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+'@sap-ux-private/control-property-editor-common': minor
+'@sap-ux-private/preview-middleware-client': minor
+'@sap-ux/control-property-editor': minor
+'@sap-ux/preview-middleware': patch
```
Comment (zdravko-georgiev) [inline .changeset/dull-spoons-clap.md:5]: Thanks for adding the `preview- middleware`. Please make all `patch`. Sorry for not being clearer with my previous commit.

**Review by zdravko-georgiev — CHANGES_REQUESTED** [review]

**Comment by zdravko-georgiev** [comment]
@nikmace Can you please also fix the code smell?

File: .changeset/dull-spoons-clap.md (hunk 10 above)
Comment (nikmace) [reply .changeset/dull-spoons-clap.md]: Changeed in: https://github.com/SAP/open-ux-tools/pull/1422/commits/d04ce0cec87bb5338752c7eec2f487602623febe

**Comment by nikmace** [comment]
> @nikmace Can you please also fix the code smell?

Yes, I am trying to fix the code smell for the 2nd day. However, I am having issues with Fiori applications in CPE. The problem is when you click for the first time on the chevron icon (in the outline tree) on any item -> all the nodes will collapse (not the one you clicked on). This behavior occurs only for Fiori applications and works fine for Adaptation Projects. I do not understand what is the issue there, the code does not have some drastic changes for that bug to appear. If you have time could you assist here?

**Comment by zdravko-georgiev** [comment]
> > @nikmace Can you please also fix the code smell?
> 
> Yes, I am trying to fix the code smell for the 2nd day. However, I am having issues with Fiori applications in CPE. The problem is when you click for the first time on the chevron icon (in the outline tree) on any item -> all the nodes will collapse (not the one you clicked on). This behavior occurs only for Fiori applications and works fine for Adaptation Projects. I do not understand what is the issue there, the code does not have some drastic changes for that bug to appear. If you have time could you assist here?

Calling in teams

**Comment by nikmace** [comment]
Update: I will be removing changes for `Tree.tsx` as this is not in the scope of this task (for less confusion and also because of a [bug](https://github.com/SAP/open-ux-tools/pull/1422#issuecomment-1798203345)). This task should only introduce changes to the redux store.

**Review by zdravko-georgiev — APPROVED** [review]
Code is well structured and easy to understand
Review comments addressed
Excellent test coverage
Manual testing ✅

**Review by voicis — APPROVED** [review]
Code changes looks good and are easy to understand
Test coverage is great
Did not test manually

**Pull request merged**